content/introduction/how-ipfs-works.md

```diff
@@ -0,0 +1,82 @@
+
+Another great feature of a Merkle-DAG and breaking content into blocks is that if you have two similar files, they can share parts of the Merkle-DAG; ie, parts of different Merkle-DAGs can reference the same data. For example, if you copyedit a document and change a few commas here and there, IPFS knows to only create new blocks for those small changes. Your old version and your new version can refer to the same blocks for everything else. This can make transferring versions of large datasets (such as genomics research or weather data) much more efficient because you only need to transfer the parts that are new or have changed instead of creating entirely new files each time.
+
+
+### 3 \\ The DHT
+
+<!-- <img alt="IPFS Stack - Routing and Exchange" src="../assets/ipfs_stack-exchange_routing.png" width="200px" /> -->
+
+So, to recap, IPFS lets you to give CIDs to content, and link that content together in a Merkle-DAG using IPLD. Now let’s move on to the last piece: how you find and move that content around!
+
+To find which peers are hosting the content you’re after (_discovery_), IPFS uses a [_distributed hash table_](https://en.wikipedia.org/wiki/Distributed_hash_table), or DHT. A hash table is a database of keys to values. A _distributed_ hash table is one where the table is split across all the peers in a distributed network. To find content, you ask these peers.
+
+The <a hrefm src="https://libp2p.io/">libp2p project</a> is the part of the IPFS ecosystem that provides the DHT and handles peers connecting and talking to each other. It can be used as a tool for other distributed systems as well; just as there is “IPFS-flavored IPLD,” there’s also “IPFS-flavored libp2p.”
+
+Once you know where your content is (ie, which peer or peers are storing the various blocks that make up the content you’re after), you use the DHT **again** to find the current location of those peers (_routing_). So, in order to get to content, you use libp2p to query the DHT twice.
+
+**TODO: —> this description of exchange might not be right, and we perhaps need something about bootstrapping, Bitswap, or Graphsync.** You’ve discovered your content, and you’ve found the current location(s) of that content — now you need to connect to that content (_exchange_). libp2p again to the rescue! _[Maybe something here about bootstrapping, Bitswap, or Graphsync.]_
```

Comment:
bootstrapping happens earlier (finding peers), but something about content transfer strategies makes sense. Even just "the default way to fetch content you want is using bitswap. After you connect to a peer that has the content you want and send them your want list (a list of the blocks you want), they will start sending you blocks of data in return using bitswap. If they have many blocks you want, they may open up a bitswap session to send you many of the blocks you want. There are other replication protocols in the works too - the most developed of which is GraphSync - which instead of asking for each specific block allows a request to be much more flexible by passing an IPLD selector describing many blocks they want all at once."  (useful resources: https://github.com/ipfs/specs/tree/master/bitswap#api-spec, https://github.com/ipfs/camp/blob/master/DEEP_DIVES/24-replication-protocol.md, https://github.com/ipld/specs/blob/master/block-layer/graphsync/graphsync.md)

Reply:
@hannahhoward - who is the actual graphsync/bitswap expert and can make sure my high level description isn't leading anyone astray

Reply:
I took these words and words from a different comment from @hsanjuan to add a section on Bitswap in https://github.com/ipfs/docs/pull/170/commits/9995c72023406277ca458947d1a6627efe55f12d -- thank you!